File: core/src/main/java/at/shockbytes/corey/common/core/workout/model/Exercise.java
package at.shockbytes.corey.common.core.workout.model;

import android.content.Context;
import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;

import at.shockbytes.corey.common.core.R;


/**
 * @author Martin Macheiner
 *         Date: 20.03.2015.
 */
public class Exercise implements Parcelable {

    protected String name;
    protected int repetitions;

    public Exercise() {
        this("", 0);
    }

    public Exercise(String name) {
        this(name, 0);
    }

    public Exercise(String name, int repetitions) {
        setName(name);
        this.repetitions = repetitions;
    }

    protected Exercise(Parcel in) {
        name = in.readString();
        repetitions = in.readInt();
    }

    public static final Creator<Exercise> CREATOR = new Creator<Exercise>() {
        @Override
        public Exercise createFromParcel(Parcel in) {
            return new Exercise(in);
        }

        @Override
        public Exercise[] newArray(int size) {
            return new Exercise[size];
        }
    };

    public void setRepetitions(int repetitions) {
        this.repetitions = repetitions;
    }

    public int getRepetitions() {
        return repetitions;
    }

    public String getDisplayName(@NonNull Context context) {

        if (repetitions == 0) {
            return name.replaceAll("_", " ");
        }
        return context.getString(R.string.format_exercise, repetitions, name).replaceAll("_", " ");
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "Exercise:\t\t" + name + "\nRepetetions:\t" + repetitions;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(name);
        parcel.writeInt(repetitions);
    }

}

File: core/src/main/java/at/shockbytes/corey/common/core/workout/model/Exercise.kt
package at.shockbytes.corey.common.core.workout.model

import android.content.Context
import android.os.Parcel
import android.os.Parcelable
import at.shockbytes.corey.common.core.R
import at.shockbytes.corey.common.core.Sortable

/**
 * Author:  Martin Macheiner
 * Date:    20.03.2015
 */
open class Exercise(
    open var name: String = "",
    open var repetitions: Int = 0,
    open var equipment: Equipment = Equipment.BODYWEIGHT
) : Parcelable, Sortable {

    constructor(parcel: Parcel) : this(
            parcel.readString() ?: "",
            parcel.readInt(),
            parcel.readSerializable() as Equipment)

    open fun getDisplayName(context: Context): String {
        return if (repetitions == 0) {
            name.replace("_".toRegex(), " ")
        } else context.getString(R.string.format_exercise, repetitions, name).replace("_".toRegex(), " ")
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeInt(repetitions)
        parcel.writeSerializable(equipment)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Exercise> {
        override fun createFromParcel(parcel: Parcel): Exercise {
            return Exercise(parcel)
        }

        override fun newArray(size: Int): Array<Exercise?> {
            return ArrayList<Exercise>(size).toTypedArray()
        }
    }
}
